gems/sorbet-runtime/test/types/props/serializable.rb

```diff
@@ -1280,6 +1280,9 @@ class ComplexStruct < T::Struct
     prop :deprecated_enum, Symbol, enum: %i[foo_one foo_two]
     prop :nilable_deprecated_enum, T.nilable(Symbol), enum: %i[foo_one foo_two]
     prop :default_deprecated_enum, Symbol, enum: %i[foo_one foo_two], default: :foo_one
+    prop :int_or_enum, T.any(MyEnum, Integer)
+    prop :int_or_str_or_enum, T.any(MyEnum, Integer, String)
+    prop :nilable_int_or_enum, T.nilable(T.any(MyEnum, Integer))
```

Comment:
add a test `T.any(MyEnum, String)`
add a test when passing the non-enum thing, like `0` or `"not an enum value"`